language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/command/executors/CreateFunctionExecutor.java

```diff
@@ -112,10 +112,12 @@ public Object execute(ExecuteCommandContext context) throws LSCommandExecutorExc
         SemanticModel semanticModel = context.workspace().semanticModel(filePath.get()).orElseThrow();
 
         FunctionCallExpressionTypeFinder typeFinder = new FunctionCallExpressionTypeFinder(semanticModel);
-        Optional<TypeSymbol> returnTypeSymbol = typeFinder.typeOf(fnCallExprNode.get());
+        fnCallExprNode.get().accept(typeFinder);
+        Optional<TypeSymbol> returnTypeSymbol = typeFinder.getReturnTypeSymbol();
+        Optional<TypeDescKind> returnTypeDescKind = typeFinder.getReturnTypeDescKind();
```

Comment:
We can move this statement into else part after line 200

Reply:
Makes sense. Moved.